frappe_testing/singles_test_fixture.py

```diff
@@ -0,0 +1,105 @@
+import frappe
+
+from .test_fixture import TestFixture
+
+
+class SinglesTestFixture(TestFixture):
+    """
+    Extends the Fixture Manager for Single doctypes
+    Just as responsible; will backup the Single doc before testing, and restore afterwards
+
+    :param singles_copy: Contains the copy of the doc fields and values
+    """
+
+    def __init__(self):
+        self.singles_copy = {}
+        super().__init__()
+
+    def setUp(self, skip_fixtures=False, skip_dependencies=False):
+        """
+        Overrides TestFixture setUp. Will backup the existing doc fields,
+        and set them to their default values.
+        Then will setup the fixture fields.
+
+        Args:
+            skip_fixtures (bool): Skip the fixture creation
+            skip_dependencies (bool): Skip the dependency creation
+
+        Returns:
+            None
+        """
+
+        if not self.DEFAULT_DOCTYPE:
+            raise Exception("DEFAULT_DOCTYPE is not defined")
+
+        if not frappe.get_meta(self.DEFAULT_DOCTYPE).get("issingle"):
+            raise Exception("Doctype is not Single")
+
+        if frappe.session.user != self.TESTER_USER:
+            frappe.set_user(self.TESTER_USER)
+
+        if self.isSetUp():
+            self.duplicate = True
+            og: TestFixture = self.get_locals_obj()[self.__class__.__name__]
+            self.fixtures = getattr(og, "fixtures", frappe._dict())
+            self._dependent_fixture_instances = getattr(
+                og, "_dependent_fixture_instances", [])
+            return
+
+        # make a copy of ALL the fields
+        self.singles_copy = frappe.get_doc(self.DEFAULT_DOCTYPE).as_dict()
+
+        # reset the single doc to a 'default' state
```

Comment:
Hm, is this necessary ? 😄 
What if while testing an `AWS Settings` single doctype, the user pre-filled all values via frappe-cms / bench-cli and when the test starts it resets to defaults ? 😬

Reply:
I made it so to keep with the spirit of Unit Testing, so we can make sure that test results don't get affected by an existing data...Do you think we can remove it completely? Have it as non-default option?

Reply:
Hm, true 👍🏼 
And then, the implementing SinglesFixture classes `eg: RommanCoreTestFixture(SinglesTestFixture)` could set their own values inside their `make_fixtures` implementation 👍🏼 

What do you think of providing a flag to disable this option ? And have it enabled by default ?

Reply:
> could set their own values inside their make_fixtures implementation 👍🏼

Exactly.

> What do you think of providing a flag to disable this option?

Sounds good.